pydantic/config.py

```diff
@@ -83,6 +83,7 @@ class BaseConfig:
     validate_assignment: bool = False
     error_msg_templates: Dict[str, str] = {}  # TODO remove
     arbitrary_types_allowed: bool = False  # TODO default True, or remove
+    warn_on_undefined_types: bool = True
```

Comment:
Also, is there documentation which needs updating? (Readme's etc.?)

Reply:
yes, would be useful to add documentation for this now. Although the docs needs a lot of work for V2. 

@jimkring Please could you and something to the docs, and I'll take care of migrating it to the new documentation structure when the time comes.